Traceback (most recent call last):
  File "/opt/jupyterhub/lib/python3.8/site-packages/jupyter_cache/executors/utils.py", line 51, in single_nb_execution
    executenb(
  File "/opt/jupyterhub/lib/python3.8/site-packages/nbclient/client.py", line 1087, in execute
    return NotebookClient(nb=nb, resources=resources, km=km, **kwargs).execute()
  File "/opt/jupyterhub/lib/python3.8/site-packages/nbclient/util.py", line 74, in wrapped
    return just_run(coro(*args, **kwargs))
  File "/opt/jupyterhub/lib/python3.8/site-packages/nbclient/util.py", line 53, in just_run
    return loop.run_until_complete(coro)
  File "/usr/lib/python3.8/asyncio/base_events.py", line 616, in run_until_complete
    return future.result()
  File "/opt/jupyterhub/lib/python3.8/site-packages/nbclient/client.py", line 540, in async_execute
    await self.async_execute_cell(
  File "/opt/jupyterhub/lib/python3.8/site-packages/nbclient/client.py", line 832, in async_execute_cell
    self._check_raise_for_error(cell, exec_reply)
  File "/opt/jupyterhub/lib/python3.8/site-packages/nbclient/client.py", line 740, in _check_raise_for_error
    raise CellExecutionError.from_cell_and_msg(cell, exec_reply['content'])
nbclient.exceptions.CellExecutionError: An error occurred while executing the following cell:
------------------
# Our data
time = [0,1.0,2.0,3.0,4.0,5.0,6.0]
#speed = [0,3,7,12,20,30,45.6]
# Our data model
def poly1(b0,b1,x):
    # return y = b0 + b1*x
    poly1=b0+b1*x
    return(poly1)
# Our plotting function
import matplotlib.pyplot as plt
def make2plot(listx1,listy1,listx2,listy2,strlablx,strlably,strtitle):
    mydata = plt.figure(figsize = (10,5)) # build a square drawing canvass from figure class
    plt.plot(listx1,listy1, c='red', marker='p',linewidth=0) # basic data plot
    plt.plot(listx2,listy2, c='blue',linewidth=1) # basic model plot
#    plt.plot(listx3,listy3, c='green',linewidth=1) # basic model plot
    plt.xlabel(strlablx)
    plt.ylabel(strlably)
    plt.legend(['Data','Model'])# modify for argument insertion
    plt.title(strtitle)
    plt.show()
# Our "fitting" process
intercept = -3.0
slope = 7.0
modelSpeed = [] # empty list
for i in range(len(time)):
    modelSpeed.append(poly1(intercept,slope,time[i]))
# Plotting results
charttitle="Plot of y=b0+b1*x model and observations \n" + " Model equation: y = " + str(intercept) + " + " + str(slope) + "x"
make2plot(time,speed,time,modelSpeed,'time (sec.)','speed (m/s)',charttitle)
------------------

[0;31m---------------------------------------------------------------------------[0m
[0;31mNameError[0m                                 Traceback (most recent call last)
[0;32m<ipython-input-1-3b0760f9c765>[0m in [0;36m<module>[0;34m[0m
[1;32m     27[0m [0;31m# Plotting results[0m[0;34m[0m[0;34m[0m[0;34m[0m[0m
[1;32m     28[0m [0mcharttitle[0m[0;34m=[0m[0;34m"Plot of y=b0+b1*x model and observations \n"[0m [0;34m+[0m [0;34m" Model equation: y = "[0m [0;34m+[0m [0mstr[0m[0;34m([0m[0mintercept[0m[0;34m)[0m [0;34m+[0m [0;34m" + "[0m [0;34m+[0m [0mstr[0m[0;34m([0m[0mslope[0m[0;34m)[0m [0;34m+[0m [0;34m"x"[0m[0;34m[0m[0;34m[0m[0m
[0;32m---> 29[0;31m [0mmake2plot[0m[0;34m([0m[0mtime[0m[0;34m,[0m[0mspeed[0m[0;34m,[0m[0mtime[0m[0;34m,[0m[0mmodelSpeed[0m[0;34m,[0m[0;34m'time (sec.)'[0m[0;34m,[0m[0;34m'speed (m/s)'[0m[0;34m,[0m[0mcharttitle[0m[0;34m)[0m[0;34m[0m[0;34m[0m[0m
[0m
[0;31mNameError[0m: name 'speed' is not defined
NameError: name 'speed' is not defined

